mat/dense_test.go

```diff
@@ -2093,3 +2093,14 @@ func denseMulTransSymBench(b *testing.B, size int, rho float64) {
 		wd = &n
 	}
 }
+
+func BenchmarkDenseSum1000(b *testing.B) { denseSumBench(b, 1000, 100) }
+
+func denseSumBench(b *testing.B, size, iters int) {
+	b.StopTimer()
+	a, _ := randDense(size, 1.0, rand.NormFloat64)
+	b.StartTimer()
+	for i := 0; i < iters; i++ {
+		_ = Sum(a)
```

Comment:
one should probably store the result in a global variable, to prevent the clever optimizer from doing its job.

Reply:
ie:
```go
var denseSumBenchResult float64

func denseSumBench(b *testing.B, size int) {
    a, _ := randDense(size, 1.0, rand.NormFloat64)
    b.ResetTimer()
    for i := 0; i < b.N; i++ {
        denseSumBenchResult = Sum(a)
    }
}
```